lib/memo_wise.rb

```diff
@@ -423,6 +423,28 @@ def #{method_name}#{args_str}
           target.singleton_class.send(:private, method_name)
         end
       end
+
+      # Override [Module#instance_method](https://ruby-doc.org/core-3.0.0/Module.html#method-i-instance_method)
+      # to proxy the original `UnboundMethod#parameters` results. This proxying
+      # helps code that uses Ruby reflection over method parameters
+      def target.instance_method(symbol)
+        original_memo_wised_name = :"_memo_wise_original_#{symbol}"
+        curr_method = super
+
```

Comment:
I found the below logic a little hard to follow. Wonder what you think of a refactor something like this:

```
        super.tap do |curr_method|
          if symbol == :initialize
            curr_method.define_singleton_method(:parameters) do
              curr_method.super_method.parameters
            end
          elsif private_method_defined?(original_memo_wised_name)
            orig_super = super(original_memo_wised_name)
            curr_method.define_singleton_method(:parameters) do
              orig_super.parameters
            end
          end
        end
```

I think it's cleaner because surrounding the whole thing in a tap makes it explicit that the goal of this method is to amend `super`. It's also more clear to me this way that if the method doesn't fall into either of the cases, we don't amend the method at all.

Reply:
Sure, I'm happy to refactor this way!

Reply:
Modifed as discussed, and added comments per @JacobEvelyn below, please re-review?

Reply:
Looks great, thanks!